File: src/main/java/com/amigos/hacer/porhacer/models/UserRepository.java
package com.amigos.hacer.porhacer.models;

import lombok.NonNull;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface UserRepository extends JpaRepository<User, String> {

    //findall by id not
    List<User> findAllByIdNot(String id);


}

File: src/main/kotlin/com/amigos/amigos/repositories/UserRepository.kt
package com.amigos.amigos.repositories

import com.amigos.amigos.models.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface UserRepository : JpaRepository<User?, Long?> {
    fun findByUsername(username: String?): User?
    override fun findAll(): List<User>
    override fun findById(id: Long): Optional<User?>
}